hw/opentitan/ot_hmac.c

```diff
@@ -407,16 +407,97 @@ static void ot_hmac_writeback_digest_state(OtHMACState *s)
 {
     /* copy intermediary digest to mock HMAC operation for stop/continue
     behaviour. */
-    /* TODO: add support for SHA2-384 and SHA2-512 */
-    for (unsigned i = 0; i < 8u; i++) {
-        STORE32H(s->ctx->state.sha256.state[i], s->regs->digest + i);
+    switch (ot_hmac_get_digest_size(s->regs->cfg)) {
+    case HMAC_SHA2_256:
+        for (unsigned idx = 0; idx < 8u; idx++) {
+            STORE32H(s->ctx->state.sha256.state[idx], s->regs->digest + idx);
+        }
+        break;
+    case HMAC_SHA2_384:
+        /* Even though SHA384 only uses the first six uint64_t values of
+            the SHA512 digest, we must store them all for intermediary
+            computation. */
```

Comment:
QEMU coding-style for multi-line comments is:
```
/*
 * line 1
 * line 2
 */
```
(there are several occurrences to fix)